auditbeat/module/file_integrity/kprobes/kallsyms.go

```diff
@@ -0,0 +1,95 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build linux
+
+package kprobes
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strings"
+)
+
+const kAllSymsPath = "/proc/kallsyms"
+
+type runtimeSymbolInfo struct {
+	symbolName          string
+	isOptimised         bool
+	optimisedSymbolName string
+}
+
+// getSymbolInfoRuntime returns the runtime symbol information for the given symbolName
+// from the /proc/kallsyms file.
+func getSymbolInfoRuntime(symbolName string) (runtimeSymbolInfo, error) {
+	kAllSymsFile, err := os.Open(kAllSymsPath)
+	if err != nil {
+		return runtimeSymbolInfo{}, err
+	}
+
+	defer func() {
```

Comment:
does the linter complain with `defer kAllSymsFile.Close()`?

Reply:
my default one was but the config used in beats repo no it doesn't so I will make this `defer kAllSymsFile.Close()`